arb_os/tracing.mini

```diff
@@ -0,0 +1,116 @@
+//
+// Copyright 2021, Offchain Labs, Inc. All rights reserved.
+//
+
+use std::bytearray::ByteArray;
+use std::bytearray::bytearray_new;
+
+use std::stack::Stack;
+use std::stack::stack_new;
+use std::stack::stack_push;
+use std::stack::stack_pop;
+
+
+type EvmTracer = struct {
+    recordStack: Stack,
+}
+
+var currentEvmTracer: EvmTracer;
+
+public impure func evmTracer_init() {
+    evmTracer_clear();
+}
+
+public impure func evmTracer_clear() {
+    currentEvmTracer = struct {
+        recordStack: stack_new(),
+    };
+}
+
+public impure func evmTracer_emit() {
+    let val = unsafecast<any>(());
+    let theStack = currentEvmTracer.recordStack;
+    loop {
+        if let Some(res) = stack_pop(theStack) {
+            let (us, item) = res;
+            theStack = us;
+            val = (item, val);
+        } else {
+            asm((20000, val),) { debugprint };
+            return;
+        }
+    }
```

Comment:
If there's a non-negligible cost to this, we could just output `currentEvmTracer.recordStack` directly

Reply:
This is probably necessary, because the representation of stacks is complicated now.  (It was optimized to cram multiple stack items into a single tuple.)